x-pack/plugins/enterprise_search/server/lib/indices/fetch_indices.ts

```diff
@@ -38,6 +38,10 @@ export const mapIndexStats = (
   return {
     aliases,
     health: indexStats?.health,
+    hidden:
+      indexData.settings?.index?.hidden === undefined
```

Comment:
question: why make this optional/possibly undefined here? We always know whether it's hidden, we can just return false or true. We don't have an API contract we need to be backwards compatible with, so just adding this property as mandatory and always defined seems fine.

Reply:
Settings are optional, as is the `hidden` setting. But if the settings object is not set or the hidden flag is undefined,the index is not hidden, so we can just treat it as a boolean value where `undefined === false`. I think that's a cleaner overall solution. We're not interested in whether the setting exists, after all, we just want to know whether the index is hidden.

Reply:
I'm good with that, and agree it is cleaner. I will update. Thanks!